src/Processors/Formats/Impl/PrettyCompactBlockOutputFormat.cpp

```diff
@@ -171,6 +173,12 @@ void PrettyCompactBlockOutputFormat::writeRow(
     }
 
     writeCString(grid_symbols.bar, out);
+    if (single_numeric_value)
+    {
+        auto value = columns[0]->getFloat64(0);
```

Comment:
We can put these two lines also inside the `writeReadableNumberTip` function, so there will be less duplication.